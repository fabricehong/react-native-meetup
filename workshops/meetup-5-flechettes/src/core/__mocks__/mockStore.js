import configureStore from 'redux-mock-store';

const middlewares = [];
const mockStore = configureStore(middlewares);

export default mockStore;