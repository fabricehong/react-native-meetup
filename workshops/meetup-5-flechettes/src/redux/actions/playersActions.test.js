import mockStore from '../../core/__mocks__/mockStore';
import {
  addPlayer,
  removePlayer,
  setPlayerName, setPlayerScore,
} from './playersActions';

let store;

describe('playersActions', () => {

  beforeAll(() => {
    store = mockStore({});
  });

  test('addPlayer should create an ADD_PLAYER action with the player name', () => {
    store.dispatch(addPlayer('Jules'));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('addPlayer should create an ADD_PLAYER action with the player name', () => {
    store.dispatch(removePlayer(0));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('addPlayer should create an ADD_PLAYER action with the player name', () => {
    store.dispatch(setPlayerName(0, 'Jules'));
    expect(store.getActions()).toMatchSnapshot();
  });

  test('addPlayer should create an ADD_PLAYER action with the player name', () => {
    store.dispatch(setPlayerScore(1, 300));
    expect(store.getActions()).toMatchSnapshot();
  });

});