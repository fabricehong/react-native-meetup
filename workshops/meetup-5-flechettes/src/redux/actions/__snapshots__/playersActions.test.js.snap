// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`playersActions addPlayer should create an ADD_PLAYER action with the player name 1`] = `
Object {
  "name": "Jules",
  "type": "ADD_PLAYER",
}
`;

exports[`playersActions addPlayer should create an ADD_PLAYER action with the player name 2`] = `
Object {
  "id": 0,
  "type": "REMOVE_PLAYER",
}
`;

exports[`playersActions addPlayer should create an ADD_PLAYER action with the player name 3`] = `
Object {
  "id": "Jules",
  "name": 0,
  "type": "SET_PLAYER_NAME",
}
`;
